MatrixSDK/Space/MXSpaceService.swift

```diff
@@ -51,6 +58,23 @@ public class MXSpaceService: NSObject {
     private let processingQueue: DispatchQueue
     private let completionQueue: DispatchQueue
     
+    private var spaces: [MXSpace] = []
+    private var spacesPerId: [String : MXSpace] = [:]
+    private var parentIdsPerRoomId: [String : Set<String>] = [:]
+    private var flattenedParentIds: [String: Set<String>] = [:]
+
+    private var rootSpaces: [MXSpace] = []
+    private var orphanedRooms: [MXRoom] = []
+    private var orphanedDirectRooms: [MXRoom] = []
+    
+    public var rootSpaceSummaries: [MXRoomSummary] {
+        get {
+            return rootSpaces.compactMap { space in
+                return space.summary
+            }
+        }
+    }
```

Comment:
It can be more optimised to update it once. It can be after `rootSpaces` has been set, either with a didSet or in `buildGraph(with rooms:[MXRoom])` directly.